custom-elements/custom-elements-registry/define.html

```diff
@@ -0,0 +1,178 @@
+    "font-face-src",
+    "font-face-uri",
+    "font-face-format",
+    "font-face-name",
+    "missing-glyph",
+  ];
+  validCustomElementNames.forEach(function (name) {
+    test(function () {
+      window.customElements.define(name, class {});
+    }, 'If name is ' + name + ', success');
+  });
+  invalidCustomElementNames.forEach(function (name) {
+    test(function () {
+      assert_throws(new SyntaxError, function () {
+        window.customElements.define(name, class {});
+      });
+    }, 'If name is ' + name + ', throw a SyntaxError');
+  });
+
+  // 3. If this CustomElementsRegistry contains an entry with name name,
+  // then throw a NotSupportedError and abort these steps.
+  test(function () {
+    window.customElements.define("test-define-dup-name", class {});
+  }, "test-define-dup-name 1st, success");
+  test(function () {
+    assert_throws(new NotSupportedError, function () {
+      window.customElements.define("test-define-dup-name", class {});
+    });
+  }, "test-define-dup-name 2nd, throw a NotSupportedError");
+
+  // 4. If this CustomElementsRegistry contains an entry with constructor constructor,
+  // then throw a NotSupportedError and abort these steps.
+  var testDupCtor = class {};
+  test(function () {
+    window.customElements.define("test-define-dup-ctor", testDupCtor);
+  }, "test-define-dup-ctor 1st, success");
+  test(function () {
+    assert_throws(new NotSupportedError, function () {
+      window.customElements.define("test-define-dup-ctor2", testDupCtor);
+    });
+  }, "test-define-dup-ctor 2nd, throw a NotSupportedError");
+
+  // 7.1. If extends is a valid custom element name,
+  // then throw a NotSupportedError.
+  validCustomElementNames.forEach(function (name) {
+    test(function () {
+      assert_throws(new NotSupportedError, function () {
+        window.customElements.define("test-define-extend-valid-name", class {}, { extends: name });
+      });
+    }, 'If extends is ' + name + ', throw a NotSupportedError');
+  });
+
+  // 7.2. If the element interface for extends and the HTML namespace is HTMLUnknownElement
+  // (e.g., if extends does not indicate an element definition in this specification),
+  // then throw a NotSupportedError.
+  [
+    // https://html.spec.whatwg.org/multipage/dom.html#elements-in-the-dom:htmlunknownelement
+    "bgsound",
+    "blink",
+    "isindex",
+    "multicol",
+    "nextid",
+    "spacer",
+    "elementnametobeunknownelement",
+  ].forEach(function (name) {
+    test(function () {
+      assert_throws(new NotSupportedError, function () {
+        window.customElements.define("test-define-extend-" + name, class {}, { extends: name });
+      });
+    }, 'If extends is ' + name + ', throw a NotSupportedError');
+  });
+
+  // 10. Let prototype be Get(constructor, "prototype"). Rethrow any exceptions.
+  // 11. If Type(prototype) is not Object,
+  // then throw a TypeError exception.
+  test(function () {
+    var c = class {};
+    c.prototype = "string";
```

Comment:
This will not work, because classes created with `class` have non-writable, non-configurable `.prototype` property. Instead, use `function c() { }; c.prototype = "string"`.